### ISSUE TITLE
How to build binary from source code by self ?

### ISSUE THREAD

**Issue by yuan78** [phuslu/goproxy#382]
How to build binary from source code by self ?
Still need gopatch? I don't understand well about Makefile and release script.

**Comment by jzp820927** [comment]
以前作者说过，好像挺麻烦的，还要搭建 什么环境

**Comment by phuslu** [comment]
这几天会写个 wiki 来描述.

**Comment by jasonliul** [comment]
@phuslu 
惊现社长, 哇呜

**Comment by jzp820927** [comment]
我去，终于现身了，欢呼ping。。。

**Comment by hcym** [comment]
我靠

特地注册，出个声，新的服务端不能用啊，老大

**Comment by fletchere** [comment]
感谢社长，goproxy又复活了。

**Comment by jzp820927** [comment]
不晓得现在最新版的 autorange 能不能用了

**Comment by ARwMq9b6** [comment]
@yuan78 Windows 下可以这样编译：
```batch
$ go get -u github.com/phuslu/goproxy
$ cd %GOPATH%/src/github.com/phuslu/goproxy
$ go build -ldflags="-X main.version=r201" -o goproxy.exe main.go
```

goproxy.exe 会出现在项目的根目录中。

**Comment by yuan78** [comment]
@ARwMq9b6, thanks, but your compile method does not work:

root@yuan# go get -u github.com/phuslu/goproxy
/home/yuan/mygo/src/github.com/phuslu/goproxy/main.go:21:2: local import "./httpproxy" in non-local package

It seems missing some package, even go get cannot finish

**Comment by ARwMq9b6** [comment]
@yuan78 It meant that go failed to build this package, but it doesn't matter, ignore it.

Do the left axnyway
```bash
$ cd $GOPATH/src/github.com/phuslu/goproxy
$ go build -ldflags="-X main.version=r201" -o goproxy main.go
```

**Comment by yuan78** [comment]
@ARwMq9b6 , thanks for your advice , but I still get below compile errors when run go build:

D:\mygo\src\github.com\phuslu\goproxy>go build -ldflags="-X main.version=r201" -
o goproxy.exe main.go
httpproxy\filters\auth\auth.go:10:2: cannot find package "github.com/cloudflare/
golibs/lrucache" in any of:
        C:\Go\src\github.com\cloudflare\golibs\lrucache (from $GOROOT)
        D:\mygo\src\github.com\cloudflare\golibs\lrucache (from $GOPATH)
httpproxy\handler.go:7:2: cannot find package "github.com/golang/glog" in any of
:
        C:\Go\src\github.com\golang\glog (from $GOROOT)
        D:\mygo\src\github.com\golang\glog (from $GOPATH)
main.go:18:2: cannot find package "github.com/golang/groupcache" in any of:
        C:\Go\src\github.com\golang\groupcache (from $GOROOT)
        D:\mygo\src\github.com\golang\groupcache (from $GOPATH)
httpproxy\filters\ratelimit\ratelimit.go:8:2: cannot find package "github.com/ju
ju/ratelimit" in any of:
        C:\Go\src\github.com\juju\ratelimit (from $GOROOT)
        D:\mygo\src\github.com\juju\ratelimit (from $GOPATH)
httpproxy\filters\iplist\dialer.go:16:2: cannot find package "github.com/miekg/d
ns" in any of:
        C:\Go\src\github.com\miekg\dns (from $GOROOT)
        D:\mygo\src\github.com\miekg\dns (from $GOPATH)
httpproxy\filters\vps\fetchserver.go:9:2: cannot find package "github.com/phuslu
/http2" in any of:
        C:\Go\src\github.com\phuslu\http2 (from $GOROOT)
        D:\mygo\src\github.com\phuslu\http2 (from $GOPATH)

BTW: my gopath is d:\mygo\

**Comment by ARwMq9b6** [comment]
Do
```batch
$ go get "github.com/cloudflare/golibs/lrucache"
$ go get "github.com/golang/glog"
$ go get "github.com/golang/groupcache"
$ go get "github.com/juju/ratelimit"
$ go get "github.com/miekg/dns"
$ go get "github.com/phuslu/http2"
```
Then `go build -ldflags="-X main.version=r201" -o goproxy.exe main.go`

**Comment by yuan78** [comment]
@ARwMq9b6 ， thanks a lot, now the library is enough, but I cannot find the binary in gopath root directory, only generate  a new pkg directory, now binary found

**Comment by ARwMq9b6** [comment]
@yuan78 Yes, the binary file has been located at "D:\mygo\src\github.com\phuslu\goproxy".

**Comment by yuan78** [comment]
@ARwMq9b6  Great and thanks a lot for your detailed guides! 
Yes, following your instructions, I hv successfully built the binaries both for windows and linux!
You're so professional that I think you deserve to publish an excellent wiki for goproxy binary build.
Thanks again and hv a nice day!

**Comment by shneedx** [comment]
go1.6, r544, goproxy/ outside $GOPATH,

during make, the following error message displayed,
../go/src/github.com/phuslu/net/http2/transport.go:726: res.RemoteAddr undefined (type *http.Response has no field or method RemoteAddr)

**Comment by phuslu** [comment]
@shneedx https://github.com/phuslu/goproxy/blob/wiki/HowToBuild.md

**Comment by shneedx** [comment]
thank you! it all works properly.
